src/unix/apple/disk.rs

```diff
@@ -72,44 +72,78 @@ impl DiskInner {
+            {
+                self
+                    .bsd_name
+                    .as_ref()
+                    .and_then(|name| crate::sys::inner::disk::get_disk_io(name))
+                    .unwrap_or_else(|| {
+                        sysinfo_debug!("Failed to update disk i/o stats");
+                        (0, 0)
+                    })
+            }
+            #[cfg(not(target_os = "macos"))]
+            (0, 0)
+        } else {
+            Default::default()
         };
-        #[cfg(not(target_os = "macos"))]
-        let (read_bytes, written_bytes) = (0, 0);
 
         self.old_read_bytes = self.read_bytes;
         self.old_written_bytes = self.written_bytes;
         self.read_bytes = read_bytes;
         self.written_bytes = written_bytes;
 
-        unsafe {
-            if let Some(requested_properties) = build_requested_properties(&[
-                ffi::kCFURLVolumeAvailableCapacityKey,
-                ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey,
-            ]) {
-                match get_disk_properties(&self.volume_url, &requested_properties) {
-                    Some(disk_props) => {
-                        self.available_space = get_available_volume_space(&disk_props);
-                        true
-                    }
-                    None => {
-                        sysinfo_debug!("Failed to get disk properties");
-                        false
+        let (total_space, available_space) = if refresh_kind.details() {
+            unsafe {
+                if let Some(requested_properties) = build_requested_properties(&[
+                    ffi::kCFURLVolumeTotalCapacityKey,
+                    ffi::kCFURLVolumeAvailableCapacityKey,
+                    ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey,
+                ]) {
+                    match get_disk_properties(&self.volume_url, &requested_properties) {
+                        Some(disk_props) => {
+                            (
+                                get_int_value(disk_props.inner(), DictKey::Extern(ffi::kCFURLVolumeTotalCapacityKey)).unwrap_or_default() as u64,
+                                get_available_volume_space(&disk_props),
+                            )
+                        }
+                        None => {
+                            sysinfo_debug!("Failed to get disk properties");
+                            Default::default()
+                        }
                     }
+                } else {
+                    sysinfo_debug!("failed to create volume key list, skipping refresh");
+                    Default::default()
                 }
-            } else {
-                sysinfo_debug!("failed to create volume key list, skipping refresh");
-                false
             }
-        }
+        } else {
+            Default::default()
+        };
+
+        self.total_space = total_space;
+        self.available_space = available_space;
```

Comment:
Now that I think about it, better to not update the value at all if not part of the update:

```suggestion
        if refresh_kind.details() {
            unsafe {
                if let Some(requested_properties) = build_requested_properties(&[
                    ffi::kCFURLVolumeTotalCapacityKey,
                    ffi::kCFURLVolumeAvailableCapacityKey,
                    ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey,
                ]) {
                    match get_disk_properties(&self.volume_url, &requested_properties) {
                        Some(disk_props) => {
                            self.total_space = get_int_value(disk_props.inner(), DictKey::Extern(ffi::kCFURLVolumeTotalCapacityKey)).unwrap_or_default() as _;
                            self.available_space = get_available_volume_space(&disk_props) as _;
                        }
                        None => {
                            sysinfo_debug!("Failed to get disk properties");
                        }
                    }
                } else {
                    sysinfo_debug!("failed to create volume key list, skipping refresh");
                }
            }
        }
```

Reply:
Here's the situation I'm thinking about (and, FWIW, that my test covers!):

```rust
// refresh everything
let disks = Disks::new_with_refreshed_list();
let available_space_0 = disks[0].available_space();

// "time passes"

// refresh just io_usage
disks.refresh_specifics(DiskRefreshKind::new().io_usage());
let available_space_1 = disks[0].available_space();
```

With your code, `available_space_1` will equal the (stale!) `available_space_0`, i.e. return a stale value without giving the user any hint that their code has a bug.

With my code, `available_space_1` will be 0. This is also wrong, of course!... but at least it's wrong in a way that might signal to the user that they've done something wrong.

As an end-user (in the absence of `Option` fields) I would slightly prefer the latter semantics. But I don't feel super strongly about it: if you insist, I'll be happy to change it to the former.

Reply:
It's something I've been thinking about for a while. Unless returning a new struct with the new data after each refresh, there isn't really a way to really tell users that there is something wrong. Maybe returning a new struct would be the solution but that will require more discussions on how to make it as good (in terms of API and of resources usage) as possible.

So for coherency, can you not fill data with default values in the meantime please? And don't hesitate to open an issue about this. If you have ideas, I'd love to hear them but let's try to make this PR go through the finish line first. ;)

Reply:
Makes sense. Done!